### PR TITLE
Cleanup code for pre-gfx9

### PR DESCRIPTION
**Pull request by xuechen417**

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit b0327033508422cde35391d0bdc53d5f5b1853e6
<details><summary>CTS tests (Failed: 0/138020)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Ubuntu navi3x, Srdcvk</summary><ul>

<li>Passed: 35401/69018 (51.3%)</li>
<li>Failed: 0/69018 (0.0%)</li>
<li>Not Supported: 33617/69018 (48.7%)</li><li>Warnings: 0/69018 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu navi2x, Srdcvk</summary><ul>

<li>Passed: 35424/69002 (51.3%)</li>
<li>Failed: 0/69002 (0.0%)</li>
<li>Not Supported: 33578/69002 (48.7%)</li><li>Warnings: 0/69002 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/PatchCopyShader.cpp [hunk 1]
```diff
@@ -122,16 +123,8 @@ bool PatchCopyShader::runImpl(Module &module, PipelineShadersResult &pipelineSha
...
-                gfxIp.major <= 8 ? "esGsLdsSize" : "streamOutTable",
-                "streamOutInfo",
-                "streamOutWriteIndex",
-                "streamOutOffset0",
-                "streamOutOffset1",
-                "streamOutOffset2",
-                "streamOutOffset3",
-                "vertexOffset"};
+    argNames = {"globalTable",      "esGsLdsSize",      "streamOutTable",   "streamOutInfo",    "streamOutWriteIndex",
+                "streamOutOffset0", "streamOutOffset1", "streamOutOffset2", "streamOutOffset3", "vertexOffset"};
```
Comment (nhaehnle) [inline lgc/patch/PatchCopyShader.cpp:127]: Interesting, is this what clang-format came up with?

File: lgc/patch/SystemValues.cpp [hunk 2]
```diff
@@ -83,11 +83,9 @@ Value *ShaderSystemValues::getEsGsRingBufDesc() {
     // Ensure we have got the global table pointer first, and insert new code after that.
     BuilderBase builder(getInternalGlobalTablePtr()->getNextNode());
     m_esGsRingBufDesc = loadDescFromDriverTable(tableOffset, builder);
-    if (m_shaderStage != ShaderStageGeometry && m_pipelineState->getTargetInfo().getGfxIpVersion().major >= 8) {
```
Comment (nhaehnle) [inline lgc/patch/SystemValues.cpp:86]: Are you sure that the `m_shaderStage` check can be removed? Does PAL set this unconditionally now perhaps?

If in doubt, perhaps back out this one change from the PR and make it a separate change.

File: lgc/patch/VertexFetch.cpp [hunk 3]
```diff
@@ -1849,8 +1849,10 @@ bool VertexFetchImpl::needPatchA2S(const VertexInputDescription *inputDesc) cons
 
   if (inputDesc->dfmt == BufDataFormat2_10_10_10 || inputDesc->dfmt == BufDataFormat2_10_10_10_Bgra) {
     if (inputDesc->nfmt == BufNumFormatSnorm || inputDesc->nfmt == BufNumFormatSscaled ||
-        inputDesc->nfmt == BufNumFormatSint)
-      needPatch = m_lgcContext->getTargetInfo().getGfxIpVersion().major < 9;
+        inputDesc->nfmt == BufNumFormatSint) {
+      needPatch = true;
```
Comment (nhaehnle) [inline lgc/patch/VertexFetch.cpp:1853]: I think this should be `needPatch = false`.

**Review by nhaehnle — COMMENTED** [review]
I think you win "most lines removed" :) Thanks!

Nit: I think when people say pre-gfxN, they mean strictly older than gfxN. So for example in the title here, it should be pre-gfx9. At least, that's how I've seen the expression being used.

I went through it all and it mostly LGTM, but I did notice two things inline.

**Comment by xuechen417** [comment]
> Nit: I think when people say pre-gfxN, they mean strictly older than gfxN. So for example in the title here, it should be pre-gfx9. At least, that's how I've seen the expression being used.
> 
Got it. Thanks! Fix the PR title.

File: lgc/patch/PatchCopyShader.cpp (hunk 1 above)
Comment (xuechen417) [reply lgc/patch/PatchCopyShader.cpp:127]: Yes. The check-format will report error if they are lined in multiple rows as before. Should I turn off clang-format for it?

File: lgc/patch/SystemValues.cpp (hunk 2 above)
Comment (xuechen417) [reply lgc/patch/SystemValues.cpp:86]: Good catch. It is a wrong removal. `m_shaderStage` cannot be removed.

File: lgc/patch/VertexFetch.cpp (hunk 3 above)
Comment (xuechen417) [reply lgc/patch/VertexFetch.cpp:1853]: Thanks. Will fix.

**Comment by xuechen417** [comment]
V1: Correct a removal and remove `VertexFetchImpl::needPatchA2S`.

**Comment by amdvlk-admin** [comment]
### Test summary for commit e92067c157f6a317778141523e052ac54a2a0897
<details><summary>CTS tests (Failed: 0/138020)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Ubuntu navi3x, Srdcvk</summary><ul>

<li>Passed: 35401/69018 (51.3%)</li>
<li>Failed: 0/69018 (0.0%)</li>
<li>Not Supported: 33617/69018 (48.7%)</li><li>Warnings: 0/69018 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu navi2x, Srdcvk</summary><ul>

<li>Passed: 35424/69002 (51.3%)</li>
<li>Failed: 0/69002 (0.0%)</li>
<li>Not Supported: 33578/69002 (48.7%)</li><li>Warnings: 0/69002 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**